thrust/thrust/iterator/transform_iterator.h

```diff
@@ -298,8 +298,25 @@ class transform_iterator
   // See goo.gl/LELTNp
   THRUST_DISABLE_MSVC_WARNING_BEGIN(4172)
 
-  _CCCL_EXEC_CHECK_DISABLE
   _CCCL_HOST_DEVICE typename super_t::reference dereference() const
+  {
+    // TODO(bgruber): I am not sure this is the correct check here. There is also the trait
+    // thrust::detail::is_wrapped_reference that sounds fitting. Only allowing to pass through l-value references
```

Comment:
IIRC, the `device_vector` references are proxy references that `is_wrapped_reference` is designed to detect. We may need to handle these specially.